From d8d128177beaa4310687fa7facb3e46cb4f02c50 Mon Sep 17 00:00:00 2001
From: kopal <kopal@broadcom.com>
Date: Wed, 27 Nov 2013 16:29:57 +0530
Subject: [PATCH 0456/1532] 3xx_xmc: Added tmpl.dts

Also edited tmpl.dts for 2xx_xmc, to remove the non used
controller details.
[Based on SDK 3.2]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/mips/boot/dts/xlp2xx-xmc-linux-tmpl.dts | 102 -------------------
 arch/mips/boot/dts/xlp3xx-xmc-linux-tmpl.dts | 145 +++++++++++++++++++++++++++
 arch/mips/boot/dts/xlp3xx-xmc-linux.dts      |   4 +-
 3 files changed, 147 insertions(+), 104 deletions(-)
 create mode 100644 arch/mips/boot/dts/xlp3xx-xmc-linux-tmpl.dts

diff --git a/arch/mips/boot/dts/xlp2xx-xmc-linux-tmpl.dts b/arch/mips/boot/dts/xlp2xx-xmc-linux-tmpl.dts
index eef37cb..09feb9c 100644
--- a/arch/mips/boot/dts/xlp2xx-xmc-linux-tmpl.dts
+++ b/arch/mips/boot/dts/xlp2xx-xmc-linux-tmpl.dts
@@ -86,41 +86,6 @@
 			reg = <0 0x4000 0x200>;
 		};
 
-		nor_flash@1,0 {
-			compatible = "cfi-flash";
-			#address-cells = <1>;
-			#size-cells = <1>;
-			bank-width = <2>;
-			reg = <1 0 0x1000000>;
-
-			partition@0 {
-				label = "x-loader";
-				reg = <0x0 0x100000>; /* 1M */
-				read-only;
-			};
-
-			partition@100000 {
-				label = "u-boot";
-				reg = <0x100000 0x100000>; /* 1M */
-			};
-
-			partition@200000 {
-				label = "kernel";
-				reg = <0x200000 0x500000>; /* 5M */
-			};
-
-			partition@700000 {
-				label = "rootfs";
-				reg = <0x700000 0x800000>; /* 8M */
-			};
-
-			partition@f00000 {
-				label = "env";
-				reg = <0xf00000 0x100000>; /* 1M */
-				read-only;
-			};
-		};
-
 		nand_flash@0,0 {
 			compatible = "netlogic,xlp-nand";
 			#address-cells = <1>;
@@ -159,61 +124,6 @@
 			};
 		};
 
-		spi: xlp_spi@13a000 {
-			compatible = "netlogic,xlp-spi";
-			#address-cells = <1>;
-			#size-cells = <0>;
-			reg = <0 0x3a000 0x100>;
-
-			spi_nor@1 {
-				compatible = "spansion,s25sl12801";
-				#address-cells = <1>;
-				#size-cells = <1>;
-
-				reg = <1>;		/* Chip Select */
-				spi-max-frequency = <40000000>;
-
-				partition@0 {
-					label = "x-loader";
-					reg = <0x0 0x200000>; /* 2M */
-					read-only;
-				};
-
-				partition@200000 {
-					label = "kernel";
-					reg = <0x200000 0x500000>; /* 5M */
-				};
-
-				partition@700000 {
-					label = "rootfs";
-					reg = <0x700000 0x800000>; /* 8M */
-				};
-
-				partition@f00000 {
-					label = "env";
-					reg = <0xf00000 0x100000>; /* 1M */
-					read-only;
-				};
-			};
-			spi_nand@2 {
-				compatible = "spinand,mt29f";
-				#address-cells = <1>;
-				#size-cells = <1>;
-				reg = <2>;	/* SPI NAND CS */
-				spi-max-frequency = <50000000>;
-
-				partition@0 {
-					label = "firmware";
-					reg = <0x0 0x1c00000>;
-				};
-
-				partition@1 {
-					label = "kernel";
-					reg = <0x1c00000 0x0>;
-				};
-			};
-
-		};
 		gpio: xlp_gpio@34000 {
 			compatible = "netlogic,xlp-gpio";
 			reg = <0 0x34000 0x1000>;
@@ -225,18 +135,6 @@
 			interrupts = <39>;
 			interrupt-controller;
 		};
-		ethernet@2,0 {
-		        compatible = "smsc,lan9115";
-		        reg = <1 0x01200000 0x100000>;
-			reg-io-width = <4>;
-
-		        interrupts = <39>;
-		        interrupt-parent = <&gpio>;
-
-		        phy-mode = "mii";
-		        mac-address = [00 11 22 33 44 55];
-			smsc,irq-push-pull;
-		};
 	};
 
 	chosen {
diff --git a/arch/mips/boot/dts/xlp3xx-xmc-linux-tmpl.dts b/arch/mips/boot/dts/xlp3xx-xmc-linux-tmpl.dts
new file mode 100644
index 0000000..5486b33
--- /dev/null
+++ b/arch/mips/boot/dts/xlp3xx-xmc-linux-tmpl.dts
@@ -0,0 +1,145 @@
+/*
+ * XLP3XX Device Tree Source for SVP boards
+ */
+
+/dts-v1/;
+/ {
+	model = "netlogic,XLP-SVP";
+	compatible = "netlogic,xlp";
+	#address-cells = <2>;
+	#size-cells = <2>;
+
+	memory {
+		device_type = "memory";
+		reg =  <0 0x00100000 0 0x0FF00000	// 255M at 1M
+			0 0x20000000 0 0xa0000000	// 2560M at 512M
+			0 0xe0000000 0 0x40000000>;
+	};
+
+	soc {
+		#address-cells = <2>;
+		#size-cells = <1>;
+		compatible = "simple-bus";
+		ranges = <0 0  0 0x18000000  0x04000000   // PCIe CFG
+			  1 0  0 0x16000000  0x02000000>; // GBU chipselects
+
+		serial0: serial@30000 {
+			device_type = "serial";
+			compatible = "ns16550";
+			reg = <0 0x30100 0xa00>;
+			reg-shift = <2>;
+			reg-io-width = <4>;
+			clock-frequency = <133333333>;
+			interrupt-parent = <&pic>;
+			interrupts = <17>;
+		};
+		serial1: serial@31000 {
+			device_type = "serial";
+			compatible = "ns16550";
+			reg = <0 0x31100 0xa00>;
+			reg-shift = <2>;
+			reg-io-width = <4>;
+			clock-frequency = <133333333>;
+			interrupt-parent = <&pic>;
+			interrupts = <18>;
+		};
+		i2c0: ocores@32000 {
+			compatible = "opencores,i2c-ocores";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0 0x32100 0xa00>;
+			reg-shift = <2>;
+			reg-io-width = <4>;
+			clock-frequency = <32000000>;
+			interrupt-parent = <&pic>;
+			interrupts = <30>;
+		};
+		i2c1: ocores@33000 {
+			compatible = "opencores,i2c-ocores";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0 0x33100 0xa00>;
+			reg-shift = <2>;
+			reg-io-width = <4>;
+			clock-frequency = <32000000>;
+			interrupt-parent = <&pic>;
+			interrupts = <31>;
+
+			rtc@68 {
+				compatible = "dallas,ds1374";
+				reg = <0x68>;
+			};
+
+			dtt@4c {
+				compatible = "national,lm90";
+				reg = <0x4c>;
+			};
+
+			eeprom@57 {
+                                compatible = "xlpge_eeprom";
+                                reg = <0x57>;
+                        };
+		};
+		pic: pic@4000 {
+			interrupt-controller;
+			#address-cells = <0>;
+			#interrupt-cells = <1>;
+			reg = <0 0x4000 0x200>;
+		};
+
+		nand_flash@0,0 {
+			compatible = "netlogic,xlp-nand";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			reg = <0 0x39000 0x1000>;
+
+			interrupts = <43>;
+			interrupt-parent = <&pic>;
+			nand-ecc-mode = "hw";
+
+			partition@0 {
+				label = "x-loader";
+				reg = <0x0 0x80000>;
+			};
+
+			partition@1 {
+				label = "u-boot";
+				reg = <0x80000 0x100000>;
+			};
+			
+			partition@2 {
+                                label = "u-boot env";
+                                reg = <0x180000 0x020000>;
+                        };
+                        partition@3 {
+                                label = "u-boot env copy";
+                                reg = <0x1a0000 0x020000>;
+                        };
+                        partition@4 {
+                                label = "sysconfig";
+                                reg = <0x1c0000 0x040000>;
+                        };
+                        partition@5 {
+                                label = "user";
+                                reg = <0x200000 0x1fe00000>;
+                        };
+		};
+
+		gpio: xlp_gpio@34000 {
+			compatible = "netlogic,xlp-gpio";
+			reg = <0 0x34000 0x1000>;
+			#gpio-cells = <2>;
+			gpio-controller;
+
+			#interrupt-cells = <1>;
+			interrupt-parent = <&pic>;
+			interrupts = <39>;
+			interrupt-controller;
+		};
+
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200 rdinit=/sbin/init";
+	};
+};
diff --git a/arch/mips/boot/dts/xlp3xx-xmc-linux.dts b/arch/mips/boot/dts/xlp3xx-xmc-linux.dts
index 8f59d8c..3624903 100644
--- a/arch/mips/boot/dts/xlp3xx-xmc-linux.dts
+++ b/arch/mips/boot/dts/xlp3xx-xmc-linux.dts
@@ -12,7 +12,7 @@
         memory {
                 device_type = "memory";
                 reg =  <0 0x00000000 0 0x0B000000
-                        0 0x20000000 0 0x68a00000>;
+                        0 0x20000000 0 0x28a00000>;
         };
 
 
@@ -77,7 +77,7 @@
 	};
 
 	chosen {
-		bootargs = " console=ttyS0,115200 rhash_entries=1048576";
+		bootargs = " console=ttyS0,115200 rdinit=/sbin/init";
 		netlogic,cpumask = <0xffff>;
 		/*linux,initrd-start=<0x00c00000>;
 		linux,initrd-end=<0x02c00000>;*/
-- 
1.9.1

